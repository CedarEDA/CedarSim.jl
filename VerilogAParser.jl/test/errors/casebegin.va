module casebegin(p, n);

    analog function casebegin;
        input x;
        integer x;

        case(x) begin
            0: casebegin = 1;
            1: casebegin = 2;
        end
        endcase
    endfunction

    inout p, n;
    electrical p, n;

    analog begin
        I(p,n) <+ 0;
    end
endmodule
