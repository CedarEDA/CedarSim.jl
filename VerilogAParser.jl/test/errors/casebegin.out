[31m[1m╭ ERROR [VAS26]:[22m[39m[31m begin/end blocks not allowed inside case statements.[39m
[31m│[39m[90m   ╭───── casebegin.va:7[39m
[31m│[90m7[0m  [90m│[0m         case(x) begin
[31m│[0m   [90m┊[0m         [33m━━┳━[0m    [91m━━┳━━
[31m│[0m   [90m┊[0m         [91m✗ Unexpected `begin`
[31m│[0m   [90m┊[0m [33m`case` statement starts here[0m
[31m│[39m[90m   │[39m
[31m│[39m[90m   ╰───── [39m
[31m╯[39m